### PR TITLE
[Android] Fix for the CarouselView doesn't scroll to the right Position after changing the ItemSource with Loop enabled.

### LINKED ISSUE #17283: [Android] CarouselView doesn't scroll to the right Position after changing the ItemSource with Loop enabled
### Description

After the Carousel ItemSource is loaded for the first time, try to reset or assign a new ItemSource and set the Position to a specific value right after that with the Loop property enabled, the Carousel won't scroll to the right item. If we disable the Loop property, the scroll works fine. I could reproduce it on Android and iOS.

### Steps to Reproduce

1 - Create a new project .NET MAUI App
2 - Add a Carousel to the page with the Loop = True
3 - Load a list of 3+ items into the Carousel ItemSource
4 - Swipe it to the last position
5 - Replace the item source list or assign a new one. It could be the same list, but it needs to reset the ItemSource.
6 - Try to set the position again to the last item programmatically right after reloading the list.
7 - If you disable the loop it works.
 
Expected outcome:  The last item should be selected
Actual outcome: The first item will be selected, and the swipe feature will get lost

### Link to public reproduction project repository

https://github.com/l0gaw/CarouselPositionLoopIssue

### Version with bug

7.0.92

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

Android 13  and iOS 16.6

### Did you find any workaround?

Disable the Loop property.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Tamilarasan-Paranthaman**
### Root Cause
 
When reloading the ItemSource, the `CurrentItem `was set to null in the `UpdateAdapter`, which caused the `UpdateInitialPosition `method to miss calculating the `itemCount `based on the `CurrentItem`. As a result, the `LoopScale `value was incorrect in the `LoopedPosition `method, causing `centerPosition `to be set to 0 in the `GetGoToIndex `method of the **CarouselViewLoopManager** class. This led to a negative index position in the `GetGoToIndex `and prevented scrolling.
 
### Description of change
 
Updated the ItemCount based on the Carousel loop and position, ensuring proper calculations to maintain the correct looped position value when reloading the items.
 
 
### Issue Fixed
 
Fixes #17283
 
### Before Fix
 
https://github.com/user-attachments/assets/b4c11e61-139a-4f6a-8a47-0d0726007297
 
### After Fix
 
 
https://github.com/user-attachments/assets/f4313876-a361-42cc-88e4-c05c6dc2d4d2

### CODE REVIEW

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue17283.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿#if ANDROID
```
Comment (PureWeen) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue17283.cs:1]: Does it make sense to run this test on iOS and WinUI as well?

We typically let these tests run on all platforms even if the original issue is only happening on Android. This also lets us make sure that we have consistent behavior between all platforms.

If it's broken on iOS and/or WinUI then log a bug and make a note of the bug inside this test.

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue17283.cs (hunk 1 above)
Comment (Tamilarasan-Paranthaman) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue17283.cs]: Okay, I have modified the script to run the test on both iOS and WinUI. Could you please take a look and let me know?

**Review by PureWeen — CHANGES_REQUESTED** [review]
it looks like windows is crashing

Just ignore windows and log an issue and then we'll be able to merge this one!

**Comment by Tamilarasan-Paranthaman** [comment]
> it looks like windows is crashing
> 
> Just ignore windows and log an issue and then we'll be able to merge this one!

I have created a bug report for the CarouselView issue on Windows and added a note in the test. Could you please take a look into it ?

**Review by PureWeen — APPROVED** [review]

**Comment by PureWeen** [comment]
- Failing test is unrelated

**Pull request merged**